### PR TITLE
backup: allow to backup during the flashback

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: ref #13787, close https://github.com/pingcap/tidb/issues/39639.

What's Changed:

```commit-message
- Allow to backup during the flashback by passing the flashback flag.
- Allow the checksum request to get the snapshot during the flashback progress.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by HuSharp — APPROVED** [review]

File: components/backup/src/endpoint.rs [hunk 1]
```diff
@@ -330,6 +330,7 @@ impl BackupRange {
         assert!(!ctx.get_replica_read());
         let snap_ctx = SnapContext {
             pb_ctx: &ctx,
+            for_flashback: self.region.is_in_flashback,
```
[the remark this answers is not in the document]
Comment (JmPotato) [reply components/backup/src/endpoint.rs:333]: I think it's unnecessary for a built-in bool type.

**Review by YuJuncen — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Comment by JmPotato** [comment]
@sticnarf Sorry to bother you again, but could you please help merge this PR? Thx.

**Review by overvenus — APPROVED** [review]

**Pull request merged**